src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -34,162 +44,541 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
+
+                default:
+                    throw new NotImplementedException($"Unknown SymbolKind: {symbol.Kind}");
+            }
+        }
+
+        private static IEnumerable<Diagnostic> GetMisspelledWordDiagnostics(ISymbol symbol, string misspelledWord)
+        {
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Assembly:
+                    // Do not report spelling rules in assembly names for now. The spelling should be enforced
+                    // at the API level and the name of the assembly on disk isn't relevant
+                    // yield return Diagnostic.Create(AssemblyRule, Location.None, misspelledWord, symbol.Name);
+                    break;
+
+                case SymbolKind.Namespace:
+                    yield return Diagnostic.Create(NamespaceRule, symbol.Locations.First(), misspelledWord, symbol.ToDisplayString());
+                    break;
+
+                case SymbolKind.NamedType:
+                    foreach (var location in symbol.Locations)
+                    {
+                        yield return Diagnostic.Create(TypeRule, location, misspelledWord, symbol.ToDisplayString());
+                    }
+
+                    break;
+
+                case SymbolKind.Method:
+                case SymbolKind.Property:
+                case SymbolKind.Event:
+                case SymbolKind.Field:
+                    yield return Diagnostic.Create(MemberRule, symbol.Locations.First(), misspelledWord, symbol.ToDisplayString());
+                    break;
+
+                case SymbolKind.Parameter:
+                    yield return symbol.ContainingType.TypeKind == TypeKind.Delegate
+                        ? Diagnostic.Create(DelegateParameterRule, symbol.Locations.First(), symbol.ContainingType.ToDisplayString(), misspelledWord, symbol.Name)
+                        : Diagnostic.Create(MemberParameterRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), misspelledWord, symbol.Name);
+
+                    break;
+
+                case SymbolKind.TypeParameter:
+                    yield return symbol.ContainingSymbol.Kind == SymbolKind.Method
+                        ? Diagnostic.Create(MethodTypeParameterRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), misspelledWord, symbol.Name)
+                        : Diagnostic.Create(TypeTypeParameterRule, symbol.Locations.First(), symbol.ContainingSymbol.ToDisplayString(), misspelledWord, symbol.Name);
+
+                    break;
+
+                case SymbolKind.Local:
+                    yield return Diagnostic.Create(VariableRule, symbol.Locations.First(), misspelledWord, symbol.ToDisplayString());
+                    break;
+
+                default:
+                    throw new NotImplementedException($"Unknown SymbolKind: {symbol.Kind}");
+            }
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext compilationStartContext)
```

Comment:
Please move this method right below `Initialize`

Reply:
Done